Code/Framework/AzCore/AzCore/Math/Vector3.inl

```diff
@@ -30,14 +32,27 @@ namespace AZ
         ;
     }
 
+    AZ_MATH_INLINE Vector3::Vector3(const Vector2& source)
+        : m_value(Simd::Vec3::FromVec2(source.GetSimdValue()))
+    {
+    }
```

Comment:
Is this actually setting z to 0.0? This seems like it's relying on undefined behavior. In the SSE case, isn't it just going to copy whatever garbage values are in the Z bits on the Vector2 into the Vector3?

You should be able to see this by doing the following:
Vector3 a(1.0, 2.0, 3.0);
Vector2 b(a);
Vector3 c(b);
// does c.m_z = 3?

I suspect you need to add an explicit `m_z = 0.0f;` here.